numpy_financial/_financial.py

```diff
@@ -840,20 +825,40 @@ def irr(values, *, guess=None, tol=1e-12, maxiter=100, raise_exceptions=False):
     #
     # which we solve using Newton-Raphson and then reverse out the solution
     # as eirr = g - 1 (if we are close enough to a solution)
-    npv_ = np.polynomial.Polynomial(values[::-1])
-    d_npv = npv_.deriv()
-    g = 1 + guess
+    
+    g = np.roots(values)
+    IRR = np.real(g[np.isreal(g)]) - 1
```

Comment:
Nitpick: I don't like having all capital variable names, could you rename it to something meaningful?

Reply:
IRR -> internal_rate_return

Reply:
That would work. Or maybe something like `irr_` if you wanted to be brief. The rest of the code seems to be brief, although I prefer being explicit in general.